### DISCUSSION TITLE
TypeLoadException: Missing implementation of RegisterLazy

### DISCUSSION THREAD

**Discussion by NeilMacMullen** [cake-build/cake#3260] [Q&A] [answered]
### What You Are Seeing?

`CakeHost.Run() throws TypeLoadException "Method 'RegisterLazy' in type 'Cake.Frosting.Internal.TypeRegistrar' from assembly 'Cake.Frosting, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation." `

### What is Expected?
No exception

### What version of Cake are you using?
1.1.0

### Are you running on a 32 or 64 bit system?
64 bit

### What environment are you running on?  Windows? Linux? Mac?
Windows but also seems to occur on Linux

### Are you running on a CI Server?  If so, which one?
Occurs on desktop and github workflows, Both WIndows and Linux

See this PR ...
https://github.com/NeilMacMullen/Textrude/pull/111/checks?check_run_id=2065515137

### How Did You Get This To Happen? (Steps to Reproduce)

It is possible to reproduce this with a  minimal C# console line project. The trigger appears to be the inclusion of Spectre.Console 0.38.0.  If Spectre.Console is reverted to 0.37, the project runs as expected.

Program.cs
```
using Cake.Frosting;

namespace spectest
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            var h = new CakeHost();
            //crashes when Run is called
            h.Run(args);
        }
    }
}
```

Project.csproj

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Cake.Frosting" Version="1.1.0" />
    <PackageReference Include="Spectre.Console" Version="0.38.0" />
  </ItemGroup>

</Project>
```

**Comment by augustoproiete** [comment, marked as answer]
Thanks for reporting @NeilMacMullen!

This is not exactly an issue with Cake _per se_, but with a breaking change in Spectre.Console 0.38.0 as you noticed.

Cake v1.1.0 depends on [Spectre.Console 0.36.0](https://github.com/cake-build/cake/blob/v1.1.0/src/Cake.Cli/Cake.Cli.csproj#L23), thus using any higher version of Spectre.Console is subject to incompatibilities like the one you found.

The upcoming release of Cake (as of this writing, v1.2.0) [will depend on Spectre.Console 0.38.0](https://github.com/cake-build/cake/pull/3247) where this new method `RegisterLazy` is implemented, and you'll be able to use Spectre.Console 0.38.0 then.

For now, your best bet is to use Spectre.Console 0.36.0 included with Cake.Frosting, or use 0.37.0 which doesn't seem to have any breaking changes - though [it has a small bug related to ANSI output](https://github.com/cake-build/cake/pull/3093).

**Comment by NeilMacMullen** [comment]
Thanks @augustoproiete  - makes sense.
